pkg/securityscan/alert/prometheusrule.go

```diff
@@ -0,0 +1,74 @@
+package alert
+
+import (
+	"bytes"
+	"text/template"
+
+	"github.com/sirupsen/logrus"
+	meta1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	k8Yaml "k8s.io/apimachinery/pkg/util/yaml"
+
+	cisoperatorapiv1 "github.com/rancher/cis-operator/pkg/apis/cis.cattle.io/v1"
+	"github.com/rancher/wrangler/pkg/name"
+)
+
+func NewPrometheusRule(clusterscan *cisoperatorapiv1.ClusterScan, clusterscanprofile *cisoperatorapiv1.ClusterScanProfile, imageConfig *cisoperatorapiv1.ScanImageConfig) (*monitoringv1.PrometheusRule, error) {
+	configdata := map[string]interface{}{
```

Comment:
Is there anything going to be added to add the num tests total, num tests na, and num tests skipped to the Alert itself (as part of the annotations)? This should be passed into the PrometheusRule and, on seeing a new scene with different values, should trigger an update to the PrometheusRule.

Reply:
We are creating the PrometheusRule prior to the scan, so we wont have these values available. Even if we see them change and update the rule, we have discussed offline that Prometheus will not pick it up in time most likely? @aiyengar2

Reply:
Also every time the scheduled scan runs, it starts from a fresh state by resetting all the values from the status

Reply:
> We are creating the PrometheusRule prior to the scan, so we wont have these values available.

True. Omitting those from the Alert sounds good to me.

> Even if we see them change and update the rule, we have discussed offline that Prometheus will not pick it up in time most likely?

Not most likely, but it's possible we could run into weird bugs because of Prometheus not picking it up in time. I think omitting these values from the alert (as we have it now) would be best, but just wanted to confirm that that's okay.